src/interval.jl

```diff
@@ -364,6 +374,257 @@ true
 ≫(a, b) = greater_than_disjoint(a, b)
 # ≫̸(a, b) = !≫(a, b)
 
+###### Set-related Helpers #####
+
+# Edge is used to represent the two bounds of an interval seperately. This
+# allows us to sort edges across sets during `mergesets` (see below).
+abstract type AbstractEdge{T}; end
+struct Edge{T} <: AbstractEdge{T}
+    value::T
+    first::Bool
+    index::Int
+    closed::Bool
+end
```

Comment:
`Edge` seems to be an `Endpoint` with an `index`. Sorting is also possible via:
```julia
using Intervals: RightEndpoint
sort(intervals, by=RightEndpoint)
```
Let me know if there's more too this than I'm currently seeing

Reply:
Oye, I totally missed the `Endpoint` objects 😳

Reply:
> Oye, I totally missed the `Endpoint` objects 😳

That's why we do code review 😄